### PR TITLE
Fix handling thousand group with escaping characters

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1924

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs [hunk 1]
```diff
@@ -120,7 +120,7 @@ public static bool IsValidFormatArg(string formatString, CultureInfo formatCultu
                     mCount = 0;
                 }
 
-                if (formatStr[i] == 'a' || formatStr[i] == 'A')
+                if ((formatStr[i] == 'a' || formatStr[i] == 'A') && (i == 0 || formatStr[i - 1] != '\''))
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs:123]: >(i == 0 || formatStr[i - 1] != '\'' [](http://example.com/codeflow?start=68&length=35)

Is this change related to thousands grouping character and escaping?

Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs:123]: This is not issue of thousand grouping but when i test it and i found that I only want to do the am pm restrict when it is not in escaping ''. so i just fix here.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**